src/main/java/org/springframework/data/couchbase/repository/support/SimpleCouchbaseRepository.java

```diff
@@ -180,7 +182,12 @@ public void deleteAll() {
 
     ViewResult response = couchbaseOperations.queryView(query);
     for (ViewRow row : response) {
-      couchbaseOperations.remove(row.id());
+      try {
+        couchbaseOperations.remove(row.id());
+      } catch (DataRetrievalFailureException e) {
+        //ignore stale deletions
+        if (!(e.getCause() instanceof DocumentDoesNotExistException)) throw e;
```

Comment:
yes but that's the one caught just above ;)
so basically I check the cause of this `DataRetrievalFailureException` to indeed be a `DocumentDoesNotExistException` (which is a bit defensive for now, but it could be another cause)